### PR TITLE
[Feature][formats][ogg] Support read ogg format message 

### PR DESCRIPTION
**Pull request by zhilinli123**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java
Comment (TyrantLucifer) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java]: throw exception.

**Review by TyrantLucifer — DISMISSED** [review]

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java
Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java]: Revised, please review

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java]: CI pass Revised, please review thanks.
@TyrantLucifer

**Comment by zhilinli123** [comment]
@liugddx  The boss helps   the review thanks.

**Comment by zhilinli123** [comment]
> @liugddx  @hailin0  The boss helps the review thanks.

**Comment by zhilinli123** [comment]
> > @liugddx  @hailin0  @Hisoka-X  The boss helps the review thanks.

File: docs/en/connector-v2/sink/Kafka.md
Comment (TaoZex) [inline docs/en/connector-v2/sink/Kafka.md]: Do not delete previous documents, users will wonder how to use another format.

File: docs/en/connector-v2/source/kafka.md
Comment (TaoZex) [inline docs/en/connector-v2/source/kafka.md]: Same as above.

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java
Comment (TaoZex) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/ogg/OggJsonDeserializationSchema.java]: Add a note for exception.

File: seatunnel-formats/seatunnel-format-json/src/test/java/org/apache/seatunnel/format/json/ogg/OggJsonSerDeSchemaTest.java
Comment (TaoZex) [inline seatunnel-formats/seatunnel-format-json/src/test/java/org/apache/seatunnel/format/json/ogg/OggJsonSerDeSchemaTest.java]:
```suggestion
        Assertions.assertNotNull(url);
```

File: docs/en/connector-v2/formats/ogg-json.md
Comment (TaoZex) [inline docs/en/connector-v2/formats/ogg-json.md]:
```suggestion
Note: please refer to documentation about the meaning of each fields.
```

File: docs/en/connector-v2/sink/Kafka.md
Comment (zhilinli123) [reply docs/en/connector-v2/sink/Kafka.md]: done~

File: docs/en/connector-v2/formats/ogg-json.md
Comment (zhilinli123) [reply docs/en/connector-v2/formats/ogg-json.md]: Thank you for your review.  I have modified it and could you please kindly check it @TaoZex

**Comment by zhilinli123** [comment]
@ashulin 
@hailin0  Please help to review. thanks

**Comment by hailin0** [comment]
Please add e2e testcase

**Comment by hailin0** [comment]
Please check ci

**Comment by zhilinli123** [comment]
> Please check ci
predecessor pr:https://github.com/apache/incubator-seatunnel/pull/4428
At present, there are some problems in this pr. CanalToKafkaIT CI fails inexplicably
@MonsterChenzhuo Zhuo Yu is fixing this problem and I'll fix this CI error when we're done

**Comment by ic4y** [comment]
@zhilinli123 Is it possible to proceed with the completion of this PR now?

**Comment by zhilinli123** [comment]
> @zhilinli123 Is it possible to proceed with the completion of this PR now?

I will continue to finish it!

**Comment by Hisoka-X** [comment]
Tips: close/open will not retry to main CI, if your ci failed, you just need retry in your fork repository action.https://github.com/zhilinli123/incubator-seatunnel/actions/runs/6505872464/job/17670460419

**Comment by zhilinli123** [comment]
> Tips: close/open will not retry to main CI, if your ci failed, you just need retry in your fork repository action.https://github.com/zhilinli123/incubator-seatunnel/actions/runs/6505872464/job/17670460419

image
@Hisoka-X here?

**Comment by Hisoka-X** [comment]
> > Tips: close/open will not retry to main CI, if your ci failed, you just need retry in your fork repository action.https://github.com/zhilinli123/incubator-seatunnel/actions/runs/6505872464/job/17670460419
> 
> image @Hisoka-X here?

yes

**Comment by zhilinli123** [comment]
PTAL: @ic4y @hailin0 @TaoZex @Hisoka-X

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java]: Easier to time out after merge?

Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java]: I tested it and felt that I did not find a timeout, the same type of tasks are running on a container, I think to avoid the startup destruction process between different containers to save some time, if later found some timeouts I think we can increase the timeout setting,kafka e2e time can at least save more than 70% of the time

Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java]: Subsequent new Feature formats will increase the e2e time more. If each format starts an e2e class test, conflicts are more likely to occur

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java:172]: This method contains too many unrelated container starts and may not be able to use ci parallelism

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java:172]: Temporary speed improvements are expected, but will introduce instability, or we agree to use data files to emulate canal/debezium？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java
Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java]: This method is effective if you use data files uniformly to simulate canal/debezium. If you continue to use canal-server/debezium-server, centralized startup will only bring instability.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java
Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaFormatIT.java:172]: > Temporary speed improvements are expected, but will introduce instability, or we agree to use data files to emulate canal/debezium？

 +1, agreed,I will make changes.

**Comment by zhilinli123** [comment]
Unnecessary containers have been removed. Test the accuracy of the format using the generated data.

**Comment by zhilinli123** [comment]
PTAL: @hailin0 @Hisoka-X

**Review by hailin0 — DISMISSED** [review]
LGTM

**Comment by Hisoka-X** [comment]
Could you try merge from dev? I think you should trigger ci again.

**Comment by Hisoka-X** [comment]
DO NOT close/open to trigger CI again. I think you should just retry failed CI in your fork repository. https://github.com/zhilinli123/incubator-seatunnel/actions/runs/6703056699/job/18214614833

**Comment by Hisoka-X** [comment]
image

**Comment by zhilinli123** [comment]
> image

Thank you. Next time I'll do it this way

**Comment by zhilinli123** [comment]
PTAL: @Hisoka-X @hailin0

**Comment by Hisoka-X** [comment]
image
The test case seem like not passed.

**Comment by zhilinli123** [comment]
Do not merge Zeta for now. Running CI has some problems

**Comment by zhilinli123** [comment]
> image The test case seem like not passed.
image
Ci passed, please review. Thank you @hailin0 @Hisoka-X  @ic4y

**Review by Hisoka-X — COMMENTED** [review]
Overall LGTM @ic4y @hailin0 PTAL.

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
Thanks @zhilinli123 !

**Review by Hisoka-X — APPROVED** [review]
Thanks @zhilinli123 !

**Pull request merged**
